### PR TITLE
PWGHF: Add cascades to table conversion and split in different process functions

### PR DESCRIPTION
**Pull request by fgrosa**
@jgrosseo @vkucera please have a look

### CODE REVIEW

File: PWGHF/TableProducer/refitPvDummy.cxx [hunk 1]
```diff
@@ -31,14 +31,13 @@ struct HfRefitPvDummy {
...
   Produces<aod::Hf3Prongs_001> rowProng3WithCollId;
+  Produces<aod::HfCascades_001> rowCascadesWithCollId;
 
-  // process function
-  void process(aod::Collisions const& collisions,
-               aod::Hf2Prongs_000 const& rowsTrackIndexProng2,
-               aod::Hf3Prongs_000 const& rowsTrackIndexProng3,
-               aod::Tracks const&)
+  // process function for 2-prong candidates
+  void process2Prongs(aod::Collisions const& collisions,
```
Comment (vkucera) [inline PWGHF/TableProducer/refitPvDummy.cxx:37]:
```suggestion
  void process2Prongs(aod::Collisions const&,
```

File: PWGHF/TableProducer/refitPvDummy.cxx [hunk 2]
```diff
@@ -54,7 +53,15 @@ struct HfRefitPvDummy {
       // fill version 001 of candidate table, containing also the collision ID
       rowProng2WithCollId(collision.globalIndex(), rowTrackIndexProng2.prong0Id(), rowTrackIndexProng2.prong1Id(), rowTrackIndexProng2.hfflag());
     }
+  } // end of process function for 2 prong candidates
+
+  PROCESS_SWITCH(HfRefitPvDummy, process2Prongs, "Produce tables for 2-prong candidates", true);
 
+  // process function for 3-prong candidates
+  void process3Prongs(aod::Collisions const& collisions,
```
Comment (vkucera) [inline PWGHF/TableProducer/refitPvDummy.cxx:61]:
```suggestion
  void process3Prongs(aod::Collisions const&,
```

File: PWGHF/TableProducer/refitPvDummy.cxx [hunk 3]
```diff
@@ -70,7 +77,27 @@ struct HfRefitPvDummy {
       // fill version 001 of candidate table, containing also the collision ID
       rowProng3WithCollId(collision.globalIndex(), rowTrackIndexProng3.prong0Id(), rowTrackIndexProng3.prong1Id(), rowTrackIndexProng3.prong2Id(), rowTrackIndexProng3.hfflag());
     }
-  } // end of process function
+  } // end of process function for 3 prong candidates
+
+  PROCESS_SWITCH(HfRefitPvDummy, process3Prongs, "Produce tables for 3-prong candidates", true);
+
+  // process function for cascade candidates
+  void processCascades(aod::Collisions const& collisions,
```
Comment (vkucera) [inline PWGHF/TableProducer/refitPvDummy.cxx:85]:
```suggestion
  void processCascades(aod::Collisions const&,
```

File: PWGHF/TableProducer/refitPvDummy.cxx [hunk 4]
```diff
@@ -70,7 +77,27 @@ struct HfRefitPvDummy {
...
+                       aod::HfCascades_000 const& rowsTrackIndexCascade,
+                       aod::Tracks const&)
+  {
+    // dummy tables for cascade candidates
+    for (const auto& rowTrackIndexCascade : rowsTrackIndexCascade) {
+
+      // original PV information
+      auto track0 = rowTrackIndexCascade.prong0_as<aod::Tracks>();
```
Comment (vkucera) [inline PWGHF/TableProducer/refitPvDummy.cxx:93]: The casting should not be needed here, if the native index column type is already `Tracks`, no?

**Comment by jgrosseo** [comment]
Thanks! Looks good to me. No objections for Vit's comments of course.

File: PWGHF/TableProducer/refitPvDummy.cxx (hunk 4 above)
Comment (fgrosa) [reply PWGHF/TableProducer/refitPvDummy.cxx]: You are right indeed, I only kept as it was before, but now I removed it! 👍

Comment (vkucera) [reply PWGHF/TableProducer/refitPvDummy.cxx]: Thanks

**Review by vkucera — APPROVED** [review]

**Pull request merged**